src/handler/guild/channel.ts

```diff
@@ -0,0 +1,260 @@
+import { SlashCommandBuilder } from "@discordjs/builders";
+import { ChannelType } from "discord-api-types/v10";
+import { CommandInteraction, MessageEmbed } from "discord.js";
+import { getRepository } from "typeorm";
+
+import { Command, CommandData } from "../../decorators";
+import { BotError } from "../../error";
+
+import { Guild as GuildEntity } from "../../entity/Guild";
+
+export class Channel {
+    @CommandData({ type: "guild" })
+    channelData() {
+        return new SlashCommandBuilder()
+            .setName("channel")
+            .setDescription("Configure channels")
+            .setDefaultPermission(false)
+            .addSubcommand((subcommand) =>
+                subcommand
+                    .setName("add")
+                    .setDescription("Add new channel")
+                    .addStringOption((option) =>
+                        option
+                            .setName("type")
+                            .setDescription(
+                                "Which type of channel to configure"
+                            )
+                            .setRequired(true)
+                            .addChoices([
+                                ["Log", "log"],
+                                ["Verify", "verify"],
+                                ["Troop Movements", "troop_movement"],
+                                ["Facility Updates", "land_facility"],
+                                ["Command", "command"],
+                            ])
+                    )
+                    .addChannelOption((option) =>
+                        option
+                            .setName("channel")
+                            .setDescription("Which channel to use")
+                            // broken typings :/
+                            /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+                            .addChannelType(ChannelType.GuildText as any)
+                            .setRequired(true)
+                    )
+            )
+            .addSubcommand((subcommand) =>
+                subcommand
+                    .setName("remove")
+                    .setDescription("Unset a configured channel")
+                    .addStringOption((option) =>
+                        option
+                            .setName("type")
+                            .setDescription("Which type of channel to unset")
+                            .setRequired(true)
```

Comment:
why do we need to specify which type to unset? is it so that channels can be configured to use more than one type?

Reply:
I guess? It's mostly that way because that's how I set up the `add` subcommand.